### PR TITLE
feat: add single toggle all command

### PR DESCRIPTION
**Pull request by jorgerojas26**
This is useful i think for toggling terminals with just one mapping.

### CODE REVIEW

**Comment by akinsho** [comment]
@jorgerojas26 thanks for the PR, this came up a while ago, and I added a `ToggleTermOpenAll` and a `ToggleTermCloseAll` command in #48 but the issue that came up there that lead to two commands was how to behave it there was one of several terminals opened. Should the toggle close the open one or open all the remaining hidden ones. This approach seems to choose to close the terminal if any is open and re-open all. 

Tbh I don't really mind but I don't want both implementations since it's unnecessary complexity so I'm in favour of choosing your approach, but it would be good to remove the `OpenAll` and `CloseAll` commands and just add a `ToggleTermToggleAll` command. Then add that to the documentation as well so users can find it.

**Comment by jorgerojas26** [comment]
Sorry for late response, i have made the changes you requested, please check and tell me if they are ok for merging.

File: plugin/toggleterm.vim [hunk 1]
```diff
@@ -11,5 +11,6 @@ endif
 " Count is 0 by default
 command! -count -complete=shellcmd -nargs=* TermExec lua require'toggleterm'.exec_command(<q-args>, <count>)
 command! -count -nargs=* ToggleTerm lua require'toggleterm'.toggle_command(<q-args>, <count>)
-command! ToggleTermCloseAll lua require'toggleterm'.toggle_all("close")
-command! ToggleTermOpenAll lua require'toggleterm'.toggle_all("open")
+" command! ToggleTermCloseAll lua require'toggleterm'.toggle_all("close")
```
Comment (akinsho) [inline plugin/toggleterm.vim:14]: @jorgerojas26 I was thinking that this and all the other code related to the old functionality should be deleted not just commented out, can you remove the commented code here and also the function they reference e.g. `toggle_all` should be replaced with your implementation so there is only one function doing this and only one command

**Review by akinsho — CHANGES_REQUESTED** [review]
@jorgerojas26 thanks for tweaking this, I think what would be best though would be to completely remove the old toggle_all code as mentioned in my comment and replace the implementation of `toggle_all` with `toggle_all_terms`

**Comment by akinsho** [comment]
Just tested this out and it works nicely 👍🏿 . Thanks for making the changes and contributing 🚀

**Comment by frt** [comment]
I think it would be nice if the open_mapping mapping did the ToggleTermToggleAll instead of the ToggleTerm. Or if configuration to do this was in the documentation.

**Comment by akinsho** [comment]
@frt this issue is almost 2 years old so no longer relevant at all. FWIW the most recent commits of the plugin will now open and close the entire terminal view by default, so if you have 2 terminals it will close two and if you open it will open 2. So nothing in this issue relevant anymore.

**Comment by frt** [comment]
Thank you @akinsho, I was using a tag (v2.2.1) without the most recent commits. After changing the Packer configuration to use one of the last commits of the main branch it is now working like I've expected in my first comment. Thank you.

**Pull request merged**